GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -505,6 +512,11 @@ public Process GetGitProcess(string command, string workingDirectory, string dot
                     this.gitBinPath,
                     this.gvfsHooksRoot ?? string.Empty);
 
+            if (this.gitObjectsRoot != null)
+            {
+                processInfo.EnvironmentVariables["GIT_OBJECT_DIRECTORY"] = this.gitObjectsRoot;

```

Comment:
If this is only for the `PrunePacked` method can that method take the `gitObjectsRoot` variable and use it?

Reply:
@kewillford yup

And i honestly don't love it.  

Based on the feedback so far i'm thinking maybe we should extend this function to allow passing in environment variables?  That way prune packed would take in the object dir and pass it down.